src/Shared/OSSGadget.cs

```diff
@@ -1,18 +1,110 @@
         {
-            CommonInitialization.Initialize();
+            HelpText helpText = HelpText.AutoBuild(result, h =>
+            {
+                h.AddDashesToOption = true;
+                h.AutoVersion = true;
+                h.AdditionalNewLineAfterOption = false;
+                h.MaximumDisplayWidth = Console.WindowWidth;
+                h.AddPostOptionsLines(BaseProjectManager.GetCommonSupportedHelpTextLines());
+                return HelpText.DefaultParsingErrorsHandler(result, h);
+            });
+            Console.Error.Write(helpText);
+        }
+
+        /// <summary>
+        ///     Use the CommandlineParser library to get the cmd line arguments
+        /// </summary>
+        /// <typeparam name="T"> </typeparam>
+        /// <param name="args"> </param>
+        /// <returns> The Action object with the parsed options </returns>
+        protected ParserResult<T> ParseOptions<T>(string[]? args)
+        {
+            var parser = new Parser();
+            var parserResult = parser.ParseArguments<T>(args);
+            parserResult.WithNotParsed(errs => DisplayHelp(parserResult, errs));
+            return parserResult;
+        }
+
+        /// <summary>
+        ///     Restores the output stream to Console, if it was changed
+        /// </summary>
+        protected void RestoreOutput()
+        {
+            if (this.redirectConsole)
+            {
+                ConsoleHelper.RestoreConsole();
+            }
+        }
+
+        /// <summary>
+        ///     Use the OutputBuilder to select the given format and return a output builder
+        /// </summary>
+        /// <param name="format"> </param>
+        /// <returns> </returns>
+        protected IOutputBuilder SelectFormat(string? format)
+        {
+            try
+            {
+                currentOutputFormat = OutputBuilderFactory.GetOutputFormat(format ?? OutputFormat.text.ToString());
+                return OutputBuilderFactory.CreateOutputBuilder(this.currentOutputFormat);
+            }
+            catch (ArgumentOutOfRangeException)
+            {
+                Logger.Error("Invalid output format, selecting text");
+            }
+
+            currentOutputFormat = OutputFormat.text;
+            return OutputBuilderFactory.CreateDefaultOutputBuilder();
+        }
+
+        /// <summary>
+        ///     Change the tool output from the existing one to the passed in file
+        /// </summary>
+        /// <param name="outputFile"> </param>
+        protected void SelectOutput(string? outputFile)
+        {
+            // output to console or file?
+            this.redirectConsole = !string.IsNullOrEmpty(outputFile);
+            if (redirectConsole && outputFile is string outputLoc)
+            {
+                if (!ConsoleHelper.RedirectConsole(outputLoc))
+                {
+                    Logger.Error("Could not switch output from console to file");
```

Comment:
This error doesn't trigger when you pass a null and it fails to switch from console to file.

Reply:
added separate error for invalid input